simulator-spring-boot/src/main/java/org/citrusframework/simulator/service/runner/DefaultScenarioExecutorService.java

```diff
@@ -34,90 +27,92 @@
 import org.citrusframework.simulator.service.ScenarioExecutorService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.ApplicationContext;
-import org.springframework.context.event.ContextClosedEvent;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
+
+import static org.citrusframework.annotations.CitrusAnnotations.injectAll;
+import static org.citrusframework.simulator.model.ScenarioExecution.EXECUTION_ID;
+
 /**
+ * Default implementation of the {@link ScenarioExecutorService} that executed all scenarios in a synchronous way. This
+ * has the advantage that all data is promised to be persisted and all messages to be handled at the end of the
+ * simulation. However, it limits the possibility to handle intermediate messages inside a single scenario, because it
+ * blocks the executor thread as long as it is running.
+ * <p>
  * {@inheritDoc}
  */
 @Service
-public class DefaultScenarioExecutorServiceImpl implements ScenarioExecutorService {
+@ConditionalOnProperty(name = "citrus.simulator.mode", havingValue = "sync", matchIfMissing = true)
+public class DefaultScenarioExecutorService implements ScenarioExecutorService {
```

Comment:
The name and other documentation implies, that this service can be replaced by a more specific implementation. Given this condition, I do not see an obvious way to achieve this, other than bean overriding. Maybe add another conditional on an existing ScenarioExecutorService - if that is feasible?

Reply:
why you can just set `citrus.simulator.mode=whatever-i-want`, so this won't match and you're good to go. but yeah, a conditional on the interface type would do additionally.

Reply:
I've updated the javadoc as well as added some asciidoc (that will be published to the documentation).

Reply:
True. A reasonable work around. I was somehow fixed on the matchIfMissing.